networkx/algorithms/tree/branchings.py

```diff
@@ -708,6 +710,10 @@ def is_root(G, u, edgekeys):
         return H
 
 
+@nx._dispatch(
+    edge_attrs={"attr": "default", "partition": None},
```

Comment:
Is there a default value for `"partition"` that should be used in this file and `tree.mst.kruskal_mst_edges`? Maybe `EdgePartition.OPEN` or some other `EdgePartition` enum?

Reply:
The code uses `None` to indicate that no partition should be considered. And that is the default value.  It is used in `find_optimum` so that would be the place to look for the impact of the default value.

Reply:
Thanks for the quick reply!

I was exploring whether it's reasonable for backends to require that data or defaults exists for all requested attributes. This simplifies implementation and allows for more performance. For example, whether `edge_attrs={"weight": None}` could, in effect, mean that `"weight"` is a required attribute with no default. When experimenting with this (i.e., failing if not all edges have data), I only found three functions that need to add defaults: `maximum_branching` (this function), `minimum_st_edge_cut` (for "capacity"), and `edge_disjoint_paths` (for "capacity"). This function, `maximum_branching`, seems somewhat special in that is has a meaning for `None` or missing attribute data.

My takeaway is that, in practice, it is probably okay for backends to raise when given `edge_attrs={"weight": None}` if not all edges have `"weight"` attribute. We should also try to specify default values whenever possible, such as `edge_attrs={"capacity": float("inf")}`.